core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -120,7 +123,8 @@ public Solver<Solution_> buildSolver() {
                 scoreDirectorFactory.getInitializingScoreTrend(),
                 solutionDescriptor,
                 ClassInstanceCache.create()).build();
-        var terminationConfig = Objects.requireNonNullElseGet(solverConfig.getTerminationConfig(), TerminationConfig::new);
+        var terminationConfig = ofNullable(configOverride).map(SolverConfigOverride::getTerminationConfig)
+                .orElseGet(() -> Objects.requireNonNullElseGet(solverConfig.getTerminationConfig(), TerminationConfig::new));
```

Comment:
You don't know this yet, but we try to avoid `Optional` _inside_ code.

Optional is great for contracts, where it says things about nullity. But inside methods, it is pointless - null checks work just as well, and read easier. 

Don't be afraid of dealing with null directly, introduce an if statement. If you encapsulate this code in a new method, it will be more readable.